Test OpenGl methods in FakeBatcher test

karupayun opened the pull request:
Part of the work to split Batcher in BatcherInterface and
PrimitiveAssembler.
(https://github.com/karupayun/orbit/commit/1d4d264fc5f1f90c033b62bc2d0b22f45750401f).

This class is testing different things (internal implementation, picking
and shared pickable) that now are going to be in separate files. As a
first step I'm testing the internal methods that are going to be used in
OpenGl implementation instead of the PrimitiveAssembler part.

In addition I'm renaming the internal methods to the original name since
they are going to be public functions used by the PrimitiveAssembler
(Henning's suggestion).

As a side effect, CaptureWindow complains correctly that they expect
that methods to be in OpenGlBatcher. This is because CaptureWindow
should use primitive assembler there. This is already in
https://github.com/karupayun/orbit/commit/1d4d264fc5f1f90c033b62bc2d0b22f45750401f,
but for now I just added a temporal TODO.

- As little temporal detail, I add GetBatcherId() and GetNumElements()
  temporary to Batcher.

2nd step in http://b/225173189

antonrohr reviewed (approved):
I am struggling a bit to see why things are done exactly the way you did them, because I dont fully understand all the connections between the different components. But I am confident that you have a plan and the moving around you do here makes sense. 

Nit: In https://github.com/karupayun/orbit/commit/1d4d264fc5f1f90c033b62bc2d0b22f45750401f the actual rename from Batcher to PrimitiveAssembler is still missing. I think doing that rename might help me understand.

beckerhe commented on src/OrbitGl/BatcherTest.cpp:
```diff
@@ -23,6 +23,27 @@ class FakeBatcher : public OpenGlBatcher {
       : OpenGlBatcher(id, picking_manager) {}
 
+  // Auxiliary methods to simplify the addition of lines, boxes and triangles.
+  using Batcher::AddLine;
+  void AddLine(Vec2 from, Vec2 to, float z, const Color& color,
+               std::unique_ptr<PickingUserData> user_data = nullptr) {
+    Color picking_color = PickingId::ToColor(PickingType::kLine, GetNumElements(), GetBatcherId());
+    return AddLine(from, to, z, color, picking_color, std::move(user_data));
+  }
+  using Batcher::AddBox;
+  void AddBox(const Box& box, const Color& color,
+              std::unique_ptr<PickingUserData> user_data = nullptr) {
+    Color picking_color = PickingId::ToColor(PickingType::kBox, GetNumElements(), GetBatcherId());
+    return AddBox(box, {color, color, color, color}, picking_color, std::move(user_data));
+  }
+  using Batcher::AddTriangle;
+  void AddTriangle(const Triangle& triangle, const Color& color,
+                   std::unique_ptr<PickingUserData> user_data = nullptr) {
+    Color picking_color =
+        PickingId::ToColor(PickingType::kTriangle, GetNumElements(), GetBatcherId());
+    return AddTriangle(triangle, {color, color, color}, picking_color, std::move(user_data));
+  }
+
```
These shadow a function from the base class. That's not good practice. Couldn't you use a free function instead?

karupayun replied:
I renamed to an internal method Add****Helper, what do you think?

karupayun commented:
I fixed a rebased issue and force-push. Really sorry for that.

I've just added a renaming Add*** methods to Add***Helper and make 2 comments to tests in BatcherTests

beckerhe commented on src/OrbitGl/BatcherTest.cpp:
```diff
@@ -178,17 +202,20 @@ void ExpectPickableEq(const FakeBatcher& batcher, const Color& rendered_color, P
-  batcher.AddTriangle(Triangle(Vec3(0, 0, 0), Vec3(0, 1, 0), Vec3(1, 0, 0)), Color(0, 255, 0, 255),
-                      triangle_pickable);
-  batcher.AddBox(Box(Vec2(0, 0), Vec2(1, 1), 0), Color(255, 0, 0, 255), box_pickable);
+  batcher.PrimitiveAssembler::AddLine(Vec2(0, 0), Vec2(1, 0), 0, Color(255, 255, 255, 255),
+                                      line_pickable);
+  batcher.PrimitiveAssembler::AddTriangle(Triangle(Vec3(0, 0, 0), Vec3(0, 1, 0), Vec3(1, 0, 0)),
+                                          Color(0, 255, 0, 255), triangle_pickable);
+  batcher.PrimitiveAssembler::AddBox(Box(Vec2(0, 0), Vec2(1, 1), 0), Color(255, 0, 0, 255),
```
The `PrimitiveAssembler::` should not be needed anymore.

beckerhe reviewed (approved):
Otherwise LGTM

karupayun commented:
Making it draft, this PR is not very clear and I will make a big PR instead.

karupayun commented:
Closing this PR as https://github.com/google/orbit/pull/3539 will be merged.
